generators/server/templates/pom.xml.ejs

```diff
@@ -146,7 +146,7 @@
         <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
         <maven-javadoc-plugin.version>3.3.1</maven-javadoc-plugin.version>
         <maven-eclipse-plugin.version>2.10</maven-eclipse-plugin.version>
-        <maven-enforcer-plugin.version>3.0.0</maven-enforcer-plugin.version>
+        <maven-enforcer-plugin.version>3.0.0-M3</maven-enforcer-plugin.version>
```

Comment:
Can you point a bug report to show the error.
Otherwise we should keep the release version.

Reply:
Hi,
While I was experiencing problems with the build, I found the issue below
and the recommendation of switching to 3.0.0-M3 solved the build error.

[#16318](https://github.com/jhipster/generator-jhipster/issues/16318)

Reply:
This may be resolved already.
There is nothing in the CI that indicates the problem.
What do you think @DanielFran

Reply:
No, did not identify also any problem related with the versions of the maven plugin.

Reply:
> This may be resolved already. There is nothing in the CI that indicates the problem. What do you think @DanielFran

Was it fixed in the POM?
Because the missing jar has not been resolved on maven central
mn.novelsoft:addressing:jar:0.0.1-SNAPSHOT

group mn doesn't exist
https://repo1.maven.org/maven2/mn/

Reply:
I can't recreate the build error by setting the version back to version 3.0.0 in a generated project.
I also tested a new install on a separate Windows machine which never had jhipster installed previously and was not able to recreate the dependency issue with enforcer-3.0.0.
Perhaps some transitive dependency has been corrected on maven central.

So I will need to undo the POM change add another commit, then squash the commits, and push the change.
Or perhaps it might just be easier to redo a new PR and abandon this one.

Reply:
No need.
```suggestion
        <maven-enforcer-plugin.version>3.0.0</maven-enforcer-plugin.version>
```